### PR TITLE
Test clang builds

### PR DESCRIPTION
**Pull request by tehnick**

### CODE REVIEW

File: tests/travis/build-and-test [hunk 1]
```diff
@@ -12,6 +12,12 @@ if [ -n "$CXX" ]; then
     QMAKE_ARGS="$QMAKE_ARGS QMAKE_CXX=$CXX"
 fi
 
+if [ "$CXX" = "clang++" ]; then
+    QMAKE_ARGS="$QMAKE_ARGS QMAKE_LINK=clang++"
+    QMAKE_ARGS="$QMAKE_ARGS QMAKE_LINK_SHLIB=clang++"
+    QMAKE_ARGS="$QMAKE_ARGS QMAKE_LFLAGS+=--coverage"
```
Comment (0xd34df00d) [inline tests/travis/build-and-test]: Does `--coverage` option really make sense in our setup?

Comment (tehnick) [inline tests/travis/build-and-test]: I haven't found another way to fix linking error.

Comment (jlaine) [inline tests/travis/build-and-test]: We don't actually use the coverage for now, so we can stop using it during test runs.

Comment (tehnick) [inline tests/travis/build-and-test]: Ok. Then do you have any idea how to disable coverage in this shell script?

Comment (jlaine) [inline tests/travis/build-and-test]: The latest code in master no longer uses coverage

Comment (tehnick) [inline tests/travis/build-and-test]: Great. I'll check and update pull request.

**Comment by tehnick** [comment]
@jlaine I have updated pull request. Please recheck it.

**Comment by tehnick** [comment]
I did Git rebase to current master branch.
Please accept or close this pull request.

**Comment by tehnick** [comment]
PR have been updated to be suitable for merging with current master branch.

**Pull request merged**